components/brave_extension/extension/brave_extension/background/api/shieldsAPI.ts

```diff
@@ -163,24 +163,34 @@ export const setAllowScriptOriginsOnce = (origins: Array<string>, tabId: number)
   })
 
 export type GetViewPreferencesData = {
-  showAdvancedView: boolean
+  showAdvancedView?: boolean
+  statsBadgeVisible?: boolean
```

Comment:
Why did you make these optional? As far as I can see, `getViewPreferences` always returns all properties, so callers won't have to do extra type checks.